core/src/main/java/org/apache/spark/memory/MemoryConsumer.java

```diff
@@ -74,26 +78,26 @@ public void spill() throws IOException {
   public abstract long spill(long size, MemoryConsumer trigger) throws IOException;
 
   /**
-   * Acquire `size` bytes memory.
+   * Acquire `size` bytes of on-heap execution memory.
    *
    * If there is not enough memory, throws OutOfMemoryError.
    */
   protected void acquireMemory(long size) {
-    long got = taskMemoryManager.acquireExecutionMemory(size, this);
+    long got = taskMemoryManager.acquireExecutionMemory(size, MemoryMode.ON_HEAP, this);
```

Comment:
it's a little confusing that this method only supports ON_HEAP. Maybe rename the method?